### DISCUSSION TITLE
if ref is defined both inside and outside a component, solidjs can't create it correctly.

### DISCUSSION THREAD

**Discussion by shi-yan** [solidjs/solid#2018] [Q&A]
I have two components: App and Sidebar

App is the parent component containing a Sidebar, the sidebar has a ref.

```jsx
    <AppContextProvider>
      <div ref={app} class={styles.App}>
        <SidePanel ref={sidePanel}></SidePanel>       
      </div>
    </AppContextProvider>
```

Inside Sidebar:

```jsx
<div ref={sidePanel}>
</div>
```

inside the Sidebar, I create another ref. 

This will result in App's sidePanel being undefined. 

How should I handle this scenario?

**Comment by atk** [comment]
Only dom elements have support for ref in solid.

**Reply by shi-yan** [reply]
Thank you. Any way to workaround this? I feel that this is a bit limiting.
